bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -164,10 +240,33 @@ public void setReconnectStrategy(AbstractReconnectStrategy reconnectStrategy) {
         reconnectStrategy.setBrokerConnection(this);
     }
 
+    /**
+     * @return Return the reconnect strategy
+     */
     public @Nullable AbstractReconnectStrategy getReconnectStrategy() {
         return this.reconnectStrategy;
     }
 
+    /**
+     * Set a timeout executor. If none is set, you will not be notified of connection timeouts, this
+     * also includes a non-firing reconnect strategy. The default executor is none.
+     *
+     * @param executor One timer will be created when a connection attempt happens
+     * @param timeoutInMS Timeout in milliseconds
+     */
+    public void setTimeoutExecutor(@Nullable ScheduledExecutorService executor, int timeoutInMS) {
+        timeoutExecutor = executor;
+        this.timeout = timeoutInMS;
+    }
+
+    public @Nullable Executor getTimeoutExecutor() {
+        return timeoutExecutor;
+    }
+
+    public int getTimeout() {
```

Comment:
The one that calls `set` knows about and I assume no one other needs to get it (or can you give an example).